Sources/SwiftGodot/Core/VariantStorable.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  VariantStorable.swift
+//
+//
+//  Created by Padraig O Cinneide on 2023-10-22.
+//
+
+@_implementationOnly import GDExtension
+ 
+/// Something that can be wrapped in a Variant, either directly through `VariantRepresentable`
+/// or by converting to a `VariantRepresentable`. 
+public protocol VariantStorable {
```

Comment:
For the sake of the public API documentation, let us enumerate in this comment examples of each.